integrations/aws/client.go

```diff
@@ -128,3 +130,33 @@ func connStatus(i *integration) sdkintegrations.OptFn {
 		return sdktypes.NewStatus(sdktypes.StatusCodeError, "bad auth type"), nil
 	})
 }
+
+func connTest(i *integration) sdkintegrations.OptFn {
+	return sdkintegrations.WithConnectionTest(func(ctx context.Context, cid sdktypes.ConnectionID) (sdktypes.Status, error) {
+		vs, err := i.vars.Get(ctx, sdktypes.NewVarScopeID(cid))
+		if err != nil {
+			return sdktypes.InvalidStatus, err
+		}
+
+		cfg, err := config.LoadDefaultConfig(ctx,
+			config.WithRegion(vs.GetValueByString("Region")),
+			config.WithCredentialsProvider(
+				credentials.NewStaticCredentialsProvider(
+					vs.GetValueByString("AccessKeyID"),
+					vs.GetValueByString("SecretKey"),
+					vs.GetValueByString("Token"))),
+		)
+		if err != nil {
+			return sdktypes.InvalidStatus, err
+		}
+
+		s3Client := s3.NewFromConfig(cfg)
+
+		_, err = s3Client.ListBuckets(ctx, &s3.ListBucketsInput{})
```

Comment:
AWS is a complicated case. User permissions are granular - some people may have S3 access, others may not. Is there something more basic, like getting the user's own details or the user's IAM groups, etc? It won't ensure that the user can you each AWS API, but it will demonstrate that the auth details are valid.

Reply:
Done